### PR TITLE
Add call to action to the remaining filters chips

### PR DESCRIPTION
**Pull request by ian-leggett**
## Description of change
As we now have more filters we now have more filter chips appearing in the collection list header. I have now added call to actions on all the filter chips so you can now remove these filters in the same way as you could with the adviser filter chips.

Note: There are clear commonalities all the filter chips but these will be addressed in the next pull request.

## Test instructions
Go to the investments projects collection page and apply new filters, once the filters are applied you should be able to clear them by clicking on the chips.

### Before
Screenshot 2020-11-25 at 13 19 59

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"

- [x] Has the branch been rebased to master?
- [x] Automated tests (Any of the following when applicable: Unit, Functional or Acceptance)
- [x] Manual compatibility testing (Browsers: Chrome, Firefox, IE11, Safari)

### CODE REVIEW

File: src/apps/investments/client/projects/state.js [hunk 1]
```diff
@@ -27,29 +32,51 @@ const searchParamProps = ({
...
-  estimated_land_date_after,
-}) => ({
-  sectors: sector_descends
-    ? sectorOptions.filter((option) => sector_descends.includes(option.value))
-    : [],
-  estimated_land_date_before,
-  estimated_land_date_after,
-})
+// Gets the label or name of the selected filter
+const getFilterLabel = (metadataOptions, filterProp) =>
```
Comment (rachaelcodes) [inline src/apps/investments/client/projects/state.js:42]: Could we change the name and the comment here to reflect that it returns an array of selected filter objects (with both labels and values)?

File: src/apps/investments/client/projects/state.js [hunk 2]
```diff
@@ -27,29 +32,51 @@ const searchParamProps = ({
...
+      },
+    ],
+  }
+
   return {
     ...state[ID],
-    payload: collectionListPayload(queryProps),
-    optionMetadata: getOptionMetadata(queryProps),
+    payload: filteredQueryProps,
+    optionMetadata: { sortOptions, sectorOptions, estimatedLandDateBefore },
```
Comment (rachaelcodes) [inline src/apps/investments/client/projects/state.js:79]: Not sure, but do we need the estimatedLandDateAfter metadata too?

Comment (rachaelcodes) [reply src/apps/investments/client/projects/state.js:79]: Hm or are we using the estimatedLandDateBefore metadata at all (I think it's just the string)

File: src/apps/investments/client/projects/state.js (hunk 1 above)
Comment (ian-leggett) [reply src/apps/investments/client/projects/state.js]: Sorry we spoke about this and I forgot to commit the change. I have renamed it and updated the comment so let me know what you think.

File: src/apps/investments/client/projects/state.js (hunk 2 above)
Comment (ian-leggett) [reply src/apps/investments/client/projects/state.js]: We actually don't need it so I have removed it for now. Thanks good spot.

**Review by tomkingston — COMMENTED** [review]
The code looks good - I have a couple of minor comments. 

On more of a qa thing.... when I ran it on my machine, cancelling the date filters clears the chip but not the filter value - I imagine that this is quite likely something that needs to be changed in my implementation of the date filters, so I can take a look?

File: test/functional/cypress/specs/investments/filter-spec.js [hunk 3]
```diff
@@ -27,62 +59,92 @@ describe('Investments Collections Filter', () => {
...
   })
 
   context('when the url contains state', () => {
     before(() => {
-      cy.visit(
-        '/investments/projects/?adviser=e83a608e-84a4-11e6-ae22-56b6b6499611'
-      )
+      cy.visit(urls.investments.projects.index(), {
+        qs: {
+          adviser: 'e83a608e-84a4-11e6-ae22-56b6b6499611',
```
[the remark this answers is not in the document]
Comment (ian-leggett) [reply test/functional/cypress/specs/investments/filter-spec.js]: Good spot, I have moved these to the top of the file for now, when we do another collection list I can look at a separate file if okay.

File: src/apps/investments/client/projects/state.js [hunk 4]
```diff
@@ -27,29 +32,51 @@ const searchParamProps = ({
...
-})
+// Gets the label or name of the selected filter
+const getFilterLabel = (metadataOptions, filterProp) =>
+  metadataOptions.filter((option) => filterProp.includes(option.value))
 
 export const state2props = ({ router, ...state }) => {
+  // Get the query props if they exist
   const queryProps = qs.parse(router.location.search.slice(1))
+
+  // Get the filtered props
```
[the remark this answers is not in the document]
Comment (ian-leggett) [reply src/apps/investments/client/projects/state.js]: Now that this has been refactored I have removed the comments as it makes more sense, totally agree 👍

**Review by tomkingston — APPROVED** [review]

**Comment by ian-leggett** [comment]
> The code looks good - I have a couple of minor comments.
> 
> On more of a qa thing.... when I ran it on my machine, cancelling the date filters clears the chip but not the filter value - I imagine that this is quite likely something that needs to be changed in my implementation of the date filters, so I can take a look?

Good spot and after our discussion we agreed that we would tackle this in another PR as it effects other filters too. Once this is done we can update the tests to cover this behaviour 👍

**Review by rachaelcodes — APPROVED** [review]
👍 looks good :)

File: src/apps/investments/client/projects/state.js (hunk 1 above)
Comment (rachaelcodes) [reply src/apps/investments/client/projects/state.js]: looks good!

**Pull request merged**